crates/bevy_app/src/task_pool_plugin.rs

```diff
@@ -159,6 +159,10 @@ impl Default for TaskPoolOptions {
 impl TaskPoolOptions {
     /// Create a configuration that forces using the given number of threads.
     pub fn with_num_threads(thread_count: usize) -> Self {
+        assert!(
+            thread_count > 2,
+            "Default task pool policies require at least 3 threads"
```

Comment:
Good, but we should add a bit more explanation here. Your PR description was really helpful to me, and users should learn those things too!

Reply:
Sounds good, I'll improve this message :)